### DISCUSSION TITLE
Unhashable Type: "dict" on scan

### DISCUSSION THREAD

**Discussion by msknight** [music-assistant/support#3376] [Q&A] [answered]
I've got a series of files which were ripped from CD. There is very little in the tags, just title, album artist, album and track number. No artwork or anything else I can see. However, all the files are reporting this in the log...
ERROR (MainThread) [music_assistant.filesystem_smb] Error processing a/Airwolf/CD 2/2-08 - Hawke's Run.flac - unhashable type: 'dict' 
...and for the life of me I can't work out what the problem is.
Searches are ending up with python discussions and a search here doesn't turn up anything even for closed queries, so I'm at a loss.

**Comment by OzGav** [comment]
I would start by properly tagging one and then moving it into the advised ROOT/ARTIST/ALBUM folder structure and then see what happens

**Reply by msknight** [reply]
I have used sound converter to convert them to WAV and then back to FLAC. I then used EasyTAG to remove all tagging. I even removed the close numbers at the start of the file names. I still get..
`2024-12-28 05:51:44.978 ERROR (MainThread) [music_assistant.filesystem_smb] Error processing a/Airwolf/Season 1 Opening and Closing Extended.flac - unhashable type: 'dict'`
I also tried uploading the WAV files instead of the flac ... same error.
Maybe there is something in the tags that easytag can't see in order to remove it?

**Reply by msknight** [reply]
I've uploaded one of the files here...
https://msknight.com/example/

Even id3tools is saying there is nothing in the metadata. I can't figure this out.

**Reply by OzGav** [reply, marked as answer]
I saw the same but I opened it is Audacity and resaved it after doing a resample to 44.1. It was ok then

**Reply by msknight** [reply]
Thanks. I'll give that a try

**Reply by Grynneman** [reply]
I'm getting the same issue, but I've got hundreds of Audio Files (m4b, m4a, ogg) gathered over the years.  
I'm trying to correct them in Musicbrainz Picard, which is a laborious task, but it doesn't seem to be fixing it.  It's disappointing as other media players (Clementine, etc) and Library Managers (Quod Libet) do not seem to be having an issue with the metadata contained within the files.  I would say that the only thing different is that these are all spoken word files (radio panel shows, radio documentaries,etc).  Resampling a spoken word file to 44.1k seems like using a sledgehammer to crack a nut.

**Reply by OzGav** [reply]
So are you saying resampling one of the files fixed that file?

**Reply by Grynneman** [reply]
I have not resampled the file, because this is only a temporary solution to something that isn't an issue in other media players or library managers.

**Reply by OzGav** [reply]
Unfortunately we have very limited resources so this may not be looked at for some time especially if you are unable to assist with troubleshooting.

**Reply by Grynneman** [reply]
So, I've done some digging into this and found the issue with the file(s).  The files were part of a larger file, each track defined with a metadata tag of [CHAPTER].  When the program that split these into separate files, it added these tags to each file.  Using software such as Musicbrainz Picard or Quod Libet to edit the tags, it does not display them.  It's only by running MediaInfo that is showed this odd metadata configuration.  Running 
`ffmpeg -i "inputfile.m4b" -f ffmetadata metadata.txt`
Produced the following output (filenames amended) -
```
;FFMETADATA1
major_brand=M4A 
minor_version=512
compatible_brands=M4A isomiso2
title=Controlling the Universe
artist=John Smith
album=John Smiths Album
date=2006-08-10
track=1/4
genre=Radio Comedy
description=A brilliant thing
encoder=Lavf60.16.100
[CHAPTER]
TIMEBASE=1/1000
START=0
END=1692851
title=Chapter 1
[CHAPTER]
TIMEBASE=1/1000
START=1692851
END=1692874
title=Chapter 2
```
As you can see, the [CHAPTER] entries are part of the metadata.  
I double checked running this through Audacity, and could see that it was being stripped out when exporting.
I then used the ffmpeg command to strip out the [CHANNEL] metadata, which is a lot quicker, as I was doing a straight copy, with no conversion.

`ffmpeg -i "inputfile.m4b" -map_chapters -1 -c copy "outputfile.m4b"`

I tested the result of this workaround in Music Assistant and it worked.

**Comment by PePeLLee** [comment]
I'm downloading podcasts into smb share and then I listen them using Music Assistant. Since some time there is some issue with id3 tags.
Rescently I had issue downloading it: [PodFetch issue](https://github.com/SamTV12345/PodFetch/issues/1018) but now it's fixed. Files are downloaded correctly.
But I cannot see them in Music assistant:
`2025-01-28 11:59:13.666 WARNING (MainThread) [music_assistant.filesystem_smb] Raport o stanie Ĺwiata Dariusza Rosiaka/Raport o stanie wiata - 25 stycznia 2025/podcast.mp3 is missing ID3 tag [albumartist], using Various Artists as fallback
2025-01-28 11:59:13.683 ERROR (MainThread) [music_assistant.webserver] Error handling message: music/tracks/get_track: unhashable type: 'dict'`

**Reply by OzGav** [reply]
What version of tags are being used. MA fully supports v2.3 tags. V2.4 can be problematic or not work at all.

**Reply by PePeLLee** [reply]
No idea how to check it. On one side:
```
id3v2 -l podcast.mp3
podcast.mp3: No ID3 tag

```
on the other:
```
ffprobe -v error -show_entries format_tags -of json podcast.mp3
{
    "format": {
        "tags": {
            "album": "Raport o stanie świata",
            "artist": "Raport o stanie świata",
            "TLEN": "8764055",
            "track": "630",
            "encoded_by": "Forecast",
```

**Reply by OzGav** [reply]
It looks like you are trying to import into MA a file which is a podcast but as a music file. That isn't supported.

**Reply by PePeLLee** [reply]
Ok, I'll wait for release 2.4.0. Briefly checked beta and seems to handle it fine.
Thanks for great work.
